src/clang_parser.cpp

```diff
@@ -844,6 +844,12 @@ std::string ClangParser::get_arch_include_path()
 std::vector<std::string> ClangParser::system_include_paths()
 {
   std::vector<std::string> result;
+#ifdef SYSTEM_INCLUDE_PATHS
+  std::istringstream lines(SYSTEM_INCLUDE_PATHS);
+  std::string line;
+  while (std::getline(lines, line, ':'))
+      result.push_back(trim(line));
+#else
```

Comment:
I think it'd be better to avoid preprocessor conditionals here. Better to keep it all visible to compiler so it's less likely to bitrot and break one day

Reply:
Hmm, we can make the default define value be an empty string and condition on the value in the code?
It won't be possible to set the config knob to an empty string at this point but I guess it'd be equivalent to setting it to /noexist or any other missing directory

Reply:
Maybe you could make the string `"-"` disable system include paths and document it like that. I'm guessing there's a magical cmake way to do what you want without resorting to a magic character but explicit is probably simpler in this case

Reply:
I took a slightly different approach: the keyword "auto" is hooks into the existing code, and the default value is auto, so:
- default behaviour is exactly the same
- explicit paths like /nix/store/whatever-glibc-vers/include works
- I've also tested things like auto:/foo and /foo:auto, even if I don't expect anyone to use that.
- the fallback is still there though, so empty string will yield the default /usr/include & /usr/local/include... but my first version also did that, so unless someone actually comes and requests such a behaviour I think this is fine.